examples/lib/src/main/java/momento/client/example/MomentoCacheApplication.java

```diff
@@ -29,10 +29,33 @@ public static void main(String[] args) {
 
       System.out.println(String.format("Getting value for key=`%s`", KEY));
 
-      CacheGetResponse getResponse = simpleCacheClient.get(CACHE_NAME, KEY);
-      System.out.println(String.format("Lookup resulted in: `%s`", getResponse.status()));
-      System.out.println(
-          String.format("Looked up value=`%s`", getResponse.string().orElse("NOT FOUND")));
+      { // Java 8 style
+        final CacheGetResponse getResponse = simpleCacheClient.getAsync(CACHE_NAME, KEY).join();
+        if (getResponse instanceof CacheGetResponse.Hit) {
+          final CacheGetResponse.Hit hit = (CacheGetResponse.Hit) getResponse;
+          System.out.println("Hit! Found value: " + hit.valueString());
+        } else if (getResponse instanceof CacheGetResponse.Miss) {
+          System.out.println("Miss!");
+        } else if (getResponse instanceof CacheGetResponse.Error) {
+          final CacheGetResponse.Error error = (CacheGetResponse.Error) getResponse;
+          System.out.println("Error encountered: " + error.exception().toString());
+        } else {
+          System.out.println("Unknown response type.");
+        }
+      }
+
+      { // Java 14 style
+        final CacheGetResponse getResponse = simpleCacheClient.getAsync(CACHE_NAME, KEY).join();
+        if (getResponse instanceof CacheGetResponse.Hit hit) {
+          System.out.println("Hit! Found value: " + hit.valueString());
+        } else if (getResponse instanceof CacheGetResponse.Miss) {
+          System.out.println("Miss!");
+        } else if (getResponse instanceof CacheGetResponse.Error error) {
+          System.out.println("Error encountered: " + error.exception().toString());
+        } else {
+          System.out.println("Unknown response type.");
+        }
+      }
```

Comment:
Thanks @nand4011 !

@kvcache @malandis @pgautier404 @poppoerika wdyt of this?  It looks pretty much identical to the way we did it in dotnet so it seems like a reasonable path forward?

Reply:
I think this is state of the art until jep420 lands in (probably) v19 at which point this should be updated to:

```
switch (getResponse) {
    case CacheGetResponse.Hit hit -> System.out.println("Hit! Found value: " + hit.valueString());
    case CacheGetResponse.Miss _  -> System.out.println("Miss!");
    default -> System.out.println("Unknown response type.");
}
```

Can we still return Error? Should that be removed?

Reply:
why wouldn't we be returning Error?

Reply:
`Unavailable` etc.?

Reply:
also this says Jep420 is available as a preview in 17/18?  Might be worth including examples of that too then when we flesh out the examples @nand4011 

https://openjdk.org/jeps/420

Reply:
wouldn't that end up an exception at join()? https://github.com/momentohq/client-sdk-java/blob/main/momento-sdk/src/main/java/momento/sdk/ScsDataClient.java#L247-L248

Reply:
I'm probably forgetting or missing something.

Reply:
lol @kvcache you wrote the blog post that instigated this philosophy of surfacing errors are part of the return type instead of throwing exceptions.... I'm confused :)

Reply:
The error handling isn't all done yet but I figured with an error response I would capture all the ways the method could throw as an error.

Reply:
I know 😄 I think I'm just missing where we actually make it a friendly return value. Don't mind me, this example looks nice.

Reply:
this is just a sketch/proposal of what it will look like after the changes.

Reply:
ahh that helps me understand, thanks.

Reply:
In the parent pull request that logic is changed to return an error instead of throwing an exception: [https://github.com/momentohq/client-sdk-java/blob/31dc780424c21e7b1527e570f4ad2b0750924d89/momento-sdk/src/main/java/momento/sdk/ScsDataClient.java#L255](url)

Reply:
> Thanks @nand4011 !
> 
> @kvcache @malandis @pgautier404 @poppoerika wdyt of this? It looks pretty much identical to the way we did it in dotnet so it seems like a reasonable path forward?

It looks good to me! This patter is pretty much identical to what we have in the .NET SDK.
Thank you for getting it started, @nand4011 🙇‍♀️